### PR TITLE
TST: fix test creating invalid CategoricalBlock

### PR DESCRIPTION
**Pull request by jbrockmendel**
The incorrect tests here turned up when trying to replace Block.get_values calls with simpler alternatives.

### CODE REVIEW

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/internals/blocks.py [hunk 1]
```diff
@@ -1745,6 +1745,10 @@ def __init__(self, values, placement, ndim=None):
         values = self._maybe_coerce_values(values)
         super().__init__(values, placement, ndim)
 
+        if self.ndim == 2 and len(self.mgr_locs) != 1:
```
[the remark this answers is not in the document]
Comment (jbrockmendel) [reply pandas/core/internals/blocks.py]: either goes here or in NonconsolidateableMixin (which should be merged into ExtensionBlock anyway)

[the remark this answers is not in the document]
Comment (jreback) [reply pandas/core/internals/blocks.py]: no this is not what i mean. how can we possibly get here when passing a Categorical? and if you can't except by directly constructing a manager, then this should be an assertion error.

[the remark this answers is not in the document]
Comment (jbrockmendel) [reply pandas/core/internals/blocks.py]: > then this should be an assertion error.

sure, will update

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
k merge on green

**Pull request merged**